src/Client/ClientBaseOptimizedParts.cpp

```diff
@@ -115,7 +118,10 @@ void ClientApplicationBase::parseAndCheckOptions(OptionsDescription & options_de
             po::variable_value value(boost::any(op.value), false);
 
             const char * option;
-            if (token.contains(' '))
+            std::error_code ec;
+            if (std::filesystem::exists(std::filesystem::path{token}, ec))
```

Comment:
:+1:

Reply:
I'm thinking maybe it makes more sense to make it `std::filesystem::is_regular_file` and not `exists`, so we ignore dirs and other things

Reply:
Even better :rocket: 

But still, is it possible to confuse it by creating a file named like `SELECT 1`?

Reply:
Yes. If you create a file called `SELECT 1` and call `clickhouse 'SELECT 1'` then it will prefer the file, but I expect only people looking for trouble will do things like this.

Reply:
Hope you're right, and this is not a way for some case of injection, like

```
echo "DROP TABLE passwords" > SELECT\ 1
clickhouse "SELECT 1"
```

Reply:
Agree with the convo here that there is, a chance of injection.. i'm trying to come up with practical attack scenario for this and though it is unlikely, I guess something like this could be one: Attacker having limited access to the machine and notice a bash script that does something like `clickhouse --user $CLICKHOUSEUSER --password CLICKHOUSEPASSWORD "insert into table select * from remote('...')"` under a different user/cronjob periodically and create a file to add himself as user to DB. 

Scenario 2: Same as above but maybe from web application with some path traversal file read/file write primitives and able to see similar job above and they decide to produce such attack. 

For that purpose, i think let's make it `if (std::filesystem::exists(std::filesystem::path{token}, ec) && !token.contains(' '))` instead to only support this with filename carry no space? if user need to have space in the file name then they can use `--query-file` to make the distinction.

Reply:
I thought for a bit about going from the other side: try to parse filename as a query. If parser does not fail, don't try to look for a file.

Reply:
Ok, let's just prefer queries to files and forget about this. I don't like parsing the query because it's expensive and the query might be wrong, so if you only accept valid queries you won't show expected errors.